crates/derive/src/combine_api.rs

```diff
@@ -395,6 +415,286 @@ pub mod test {
+            // of the two actually present values, and the default for intProp has no effect.
+            (true, json!({"aKey": "a", "intProp": 3})),
+            (false, json!({"aKey": "a", "intProp": 1})),
+            (false, json!({"aKey": "a", "strProp": "something"})),
+            // The default value applies exclusively to extracted fields when none of the combined
+            // documents have a value present for the field. Here the value for the extracted field
+            // intProp is equal to the default value, but not summed from each document with the
+            // value absent. The default value is not present in the reduced document.
+            (true, json!({"aKey": "b"})),
+            (false, json!({"aKey": "b"})),
+            (false, json!({"aKey": "b", "strProp": "something"})),
+            // A field with a default value that is not included in extracted fields will never be
+            // present in the extracted fields, and as before never be included in document
+            // reduction.
+            (true, json!({"aKey": "c", "strPropNotExtracted": "first"})),
+            (false, json!({"aKey": "c", "strPropNotExtracted": "second"})),
+            (false, json!({"aKey": "c"})),
+        ];
+
+        insta::assert_debug_snapshot!(run_simple_svc(schema_json, key_ptrs, field_ptrs, &docs));
+    }
+
+    #[test]
+    fn test_combine_with_defaults_different_types() {
+        let schema_json = serde_json::json!({
+            "properties": {
+                "intProp": {
+                    "type": "integer",
+                    "default": 7,
+                },
+                "numProp": {
+                    "default": 12.4,
+                    "type": "number",
+                },
+                "strProp": {
+                    "default": "defaultString",
+                    "type": "string",
+                },
+                "boolProp": {
+                    "default": true,
+                    "type": "boolean",
+                },
+                "objProp": {
+                    "default": { "prop": "val" },
+                    "type": "object",
+                },
+                "arrayProp": {
+                    "default": [1, "hello", null],
+                    "type": "array",
+                },
+            },
+            "reduce": {"strategy": "merge"},
+        })
+        .to_string();
+
+        let key_ptrs = vec!["/aKey".to_owned()];
+
+        let field_ptrs = vec![
+            "/intProp".to_owned(),
+            "/numProp".to_owned(), // TODO: Snapshot for this doesn't work very well.
+            "/strProp".to_owned(),
+            "/boolProp".to_owned(),
+            "/objProp".to_owned(),
+            "/arrayProp".to_owned(),
+        ];
+
+        let docs = vec![(true, json!({"aKey": "a"}))];
+
+        insta::assert_debug_snapshot!(run_simple_svc(schema_json, key_ptrs, field_ptrs, &docs));
+    }
+
+    #[test]
+    fn test_combine_with_defaults_nested() {
```

Comment:
Since writing these tests, I've realized that the rest of these in particular are really testing `Shape`'s ability to return annotations from located fields. I'm considering moving/merging them into test cases for `doc/inference`.

Reply:
👍 the new service-level testing is very much appreciated, though. I could see shuffling some of the test coverage details around but certainly not a big deal.

Reply:
Yeah I think I'll actually just leave them here. I couldn't find an obvious place to put them closer to `Shape` and they are kind of nice as documentation for how the combiner handles default values, even if they are a bit redundant.